src/components/CheckIn/CheckInDetails.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+  cloneDeep,
+  find,
+  get,
+} from 'lodash';
+
+import {
+  Callout,
+  Modal,
+} from '@folio/stripes/components';
+
+import getLocationsForSelect from '../Utils/getLocationsForSelect';
+import { fetchItems } from '../Receiving/util';
+import {
+  CHECKIN,
+  ITEMS,
+  LOCATIONS,
+} from '../Utils/resources';
+import { ITEM_STATUS } from '../Receiving/const';
+import CHECKIN_URLS from './const';
+import ItemsList from './ItemsList';
+import CheckInDetailsFooter from './CheckInDetailsFooter';
+import checkInItems from './util';
+
+class CheckInDetails extends Component {
+  static manifest = Object.freeze({
+    checkIn: CHECKIN,
+    items: ITEMS,
+    locations: LOCATIONS,
+    query: {},
+  })
+
+  static propTypes = {
+    close: PropTypes.func.isRequired,
+    location: PropTypes.object,
+    mutator: PropTypes.object,
+    pieces: PropTypes.arrayOf(PropTypes.object).isRequired,
+    resources: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.callout = React.createRef();
+
+    this.state = {
+      isAllChecked: true,
+      isLoading: true,
+      items: [],
+    };
+  }
+
+  componentDidMount() {
+    const { mutator, pieces } = this.props;
+
+    fetchItems(mutator, pieces).then(itemsMap => this.setState(() => {
+      const items = cloneDeep(pieces);
+
+      const updatedItems = items.map(item => {
+        item.barcode = get(itemsMap, [item.itemId, 'barcode'], '');
+        item.itemStatus = ITEM_STATUS.received;
+
+        return item;
+      });
+
+      return {
+        isLoading: false,
+        items: updatedItems,
+      };
+    }));
+  }
+
+  toggleItem = (item) => {
+    this.setState(state => {
+      const items = cloneDeep(state.items);
```

Comment:
I'd suggest you to clone only item you selected and than build a new array with mix of old and new

Reply:
Fixed